singularity-eos/eos/eos_base.hpp

```diff
@@ -21,6 +21,15 @@
 #include <ports-of-call/portability.hpp>
 #include <ports-of-call/portable_errors.hpp>
 
+#ifdef SINGULARITY_USE_SPINER_WITH_HDF5
+#include <spiner/databox.hpp>
+namespace singularity {
+
+using DataBox = Spiner::DataBox<Real, std::true_type>;
```

Comment:
You shouldn't use `std::true_type` here. That's a SFINAE parameter. Let it resolve to the default value.

Reply:
Also it's probably better to put this type alias inside each relevant EOS class, as when @dholladay00 and I are done iwth hierarchical grids, the different EOS's might have different template parameters in their databoxes.

Reply:
I needed explicit `true_type` for some errors before you updated `spiner` with the template parameters fixes. These are now removed.

`DataBox` is now scoped to the class, save for the tests where they are declared at file-scope (in `profile_eos/eospac.cpp`).

Reply:
:+1: perfect.